src/anonymous-credentials/presentation-builder.ts

```diff
@@ -304,26 +314,44 @@ export class PresentationBuilder extends Versioned {
     const credStatusAux: [number, string, Uint8Array][] = [];
 
     const setupParamsTrk = new SetupParamsTracker();
+    const sigParamsByScheme = Object.create(null);
 
     // Reset spec state (incase this method is called more than once)
     this.spec.reset();
 
-    // Create statements and witnesses for proving possession of each credential, i.e. proof of knowledge of BBS+ sigs.
+    // Create statements and witnesses for proving possession of each credential, i.e. proof of knowledge of the sigs.
     // Also collect encoded attributes used in any predicate
     for (let i = 0; i < numCreds; i++) {
       const cred = this.credentials[i][0];
       const schema = cred.schema as CredentialSchema;
       const flattenedSchema = schema.flatten();
 
       const numAttribs = flattenedSchema[0].length;
-      if (maxAttribs < numAttribs) {
-        sigParams = sigParams.adapt(numAttribs);
-        maxAttribs = numAttribs;
-      }
       let revealedNames = this.revealedAttributes.get(i);
       if (revealedNames === undefined) {
         revealedNames = new Set();
       }
+      const paramsClass = paramsClassBySignature(cred.signature);
+      if (paramsClass == null) {
+        throw new Error(`Invalid signature: ${cred.signature}`);
+      }
+
+      sigParamsByScheme[paramsClass.name] ??= {
```

Comment:
Why not pass `num` as `numAttribs` at this point?

Reply:
Not sure what you mean. This is done this way to avoid regenerating the same default parameters for each schema on every call to this function.

Reply:
I understand but you can generate the params for `numAttribs` and not `2`

Reply:
Let's use your approach then.
When I wrote this, I guessed that it would be less wasteful to extend existing `n`-message signature params by `m` more messages than generating new params for `n + m` messages.

Reply:
The extra cost would be while making the first wasm call that will serialize and deserialize the params